### PR TITLE
xds: use typedstruct

### PR DESCRIPTION
**Pull request by kyessenov**
Use UDPA typed struct since struct is deprecated.
Follow up to https://github.com/istio/istio/pull/19899.

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: istio-control/istio-discovery/templates/telemetryv2_1.5.yaml [hunk 1]
```diff
@@ -161,20 +173,23 @@ spec:
...
+            "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+            type_url: type.googleapis.com/envoy.config.filter.http.wasm.v2.Wasm
+            value:
+              config:
+                root_id: stackdriver_outbound
+                configuration: |
+                  {{- if not .Values.telemetry.v2.stackdriver.configOverride }}
+                  {"enable_mesh_edges_reporting": {{ .Values.telemetry.v2.stackdriver.topology }}, "disable_server_access_logging": {{ not .Values.telemetry.v2.stackdriver.logging }}, "meshEdgesReportingDuration": "600s"}
+                  {{- else }}
+                  {{ toJson .Values.telemetry.v2.stackdriver.configOverride | indent 8 }}
```
Comment (kyessenov) [inline istio-control/istio-discovery/templates/telemetryv2_1.5.yaml]: qq: why is this indent 8 and the other one 16? should I bump both by two spaces?

Comment (mandarjog) [reply istio-control/istio-discovery/templates/telemetryv2_1.5.yaml]: Yes. Though best to render the config with overrides set. I am not sure if there is a test for it.

Comment (kyessenov) [reply istio-control/istio-discovery/templates/telemetryv2_1.5.yaml]: Yes meaning update all to 18?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Added do not merge incase you want to change the indent

**Pull request merged**